app/Http/Controllers/Admin/BookController.php

```diff
@@ -108,16 +111,53 @@ public function edit($id)
      * Update infomation of Book.
      *
      * @param App\Http\Requests\EditBookRequest $request form edit book
-     * @param int                               $id      id of book
+     * @param App\Model\Book                    $book    pass book object
      *
      * @return \Illuminate\Http\Response
      */
-    public function update(EditBookRequest $request, $id)
+    public function update(EditBookRequest $request, Book $book)
     {
-        dd($request);
-        dd($id);
+        //save image path, move image to directory
+        if (isset($request->picture)) {
+            $oldPath = $book->picture;
+            if (File::exists($oldPath)) {
+                File::delete($oldPath);
+            }
+            $image = $request->picture;
+            $name = $image->hashName();
+            $folder = config('image.book.path');
+            $image->move($folder, $name);
+
+            $newPath = $folder . $name;
+            $book->picture = $newPath;
+        }
+
+        //save new donator
+        $user = User::where('employ_code', $request->iddonator)->first();
```

Comment:
*Nên chia nhỏ đoạn code thành nhiều phần, mỗi phần là 1 chức năng nhỏ - nhỏ nhất có thể*.
=> Đoạn lưu data này nên viết 1 method riêng, tên `updateBook($request)`, ở dưới `if ($this->updateBook($request)) {...}`, cho cái hàm nó gọn lẹ.